doc/BGP/BGP-router-id.md

```diff
@@ -0,0 +1,92 @@
+# BGP Router ID Customized
+
+- [Revision](#revision)
+- [Definitions/Abbreviations](#definitionsabbreviations)
+- [Scope](#scope)
+- [Overview](#overview)
+- [Requirements](#requirements)
+- [High Level Design](#high-level-design)
+- [Config DB Enhancement](#config-db-enhancement)
+    - [DEVICE_METADATA](#device_metadata)
+
+### Revision
+
+| Revision | Date        | Author           | Change Description |
+| -------- | ----------- | ---------------- | ------------------ |
+| 1.0      | Mar 27 2024 | Yaqiang Zhu, Jing Kan | Initial proposal   |
+
+### Definitions/Abbreviations
+
+| Definitions/Abbreviation | Description |
+| ------------------------ | ----------- |
+| FRR | A free and open source Internet routing protocol suite for Linux and Unix platforms |
+| Router ID | 32-bit value that uniquely identifies a BGP device |
+
+### Scope
+
+This document describes a mechanism that to use customized router id when Loopback IPv4 address is not configured.
+
+### Overview
+
+Currently, there is hard coding in SONiC to use Loopback IPv4 address to be BGP router id [bgp.main.conf.j2](https://github.com/sonic-net/sonic-buildimage/blob/master/dockers/docker-fpm-frr/frr/bgpd/bgpd.main.conf.j2#L91), which would cause BGP cannot work when IPv4 address in Loopback doesn't configure.
+
+### Requirements
+
+Add support to customize BGP router id.
+
+### High Level Design
+
+1. Add field `bgp_router_id` into `CONFIG_DB["DEVICE_METADATA"]["localhost"]["bgp_router_id"]`
+2. In [bgp.main.conf.j2](https://github.com/sonic-net/sonic-buildimage/blob/master/dockers/docker-fpm-frr/frr/bgpd/bgpd.main.conf.j2#L91) to always respect bgp_router_id in CONFIG_DB if configured
```

Comment:
I don't think we need discuss detail implementation here.
Form high-level design, let's discuss different situation here:
||Loopback0 IPv4 address exists|Loopback0 IPv4 address not exists|
|---|---|---|
|bgp_router_id configured|honor bgp_router_id|honor bgp_router_id|
|bgp_router_id not configured|honor Loopback0 IPv4|FRR default router ID value is selected as the largest IP Address of the interfaces. When router zebra is not enabled _bgpd_ can’t get interface information so router-id is set to 0.0.0.0|